plugins/api/src/api.ts

```diff
@@ -0,0 +1,95 @@
+import type {
+  BasePluginArgs,
+  Bundle,
+  DatabasePlugin,
+  DatabasePluginHooks,
+  ApiPluginResponse,
+} from "@hot-updater/plugin-core";
+
+export interface IApiConfig {
+  endpoint: string;
+  headers: Record<string, string>;
+}
+
+export const api =
+  (config: IApiConfig, hooks?: DatabasePluginHooks) =>
+  (_: BasePluginArgs): DatabasePlugin => {
+    const headers = {
+      "Content-Type": "application/json",
+      ...config.headers,
+    };
+
+    let bundles: Bundle[] = [];
+
+    return {
+      name: "api",
+      async commitBundle() {
+        const response = await fetch(config.endpoint, {
+          method: "POST",
+          headers,
+          body: JSON.stringify(bundles),
+        });
```

Comment:
https://github.com/gronxb/hot-updater/blob/b0f645155ca1c8ae15248ca857ac9edd683ebae4/plugins/cloudflare/src/d1Database.ts#L28-L89

Looking here, the `d1Database` only sends actual changes to the server and performs an upsert.

The key point seems to be transmitting only the actual changes.